### PR TITLE
Flag 34 - Add REST Resource for Flag

### PR DESCRIPTION
**Pull request by maimoonak**
https://issues.openmrs.org/browse/FLAG-34

Added REST Resource for Flag with CRUD, and search. Also extended API to allow searching based on multiple param

### CODE REVIEW

**Comment by ssmusoke** [comment]
@mogoodrich  @wluyima @djazayeri Please can you take a look at this Web Service PR and advise accordingly, I am unable to review in transit - Thanks in advance 

@maimoonak Please reach out to the colleagues mentioned above they are probably in the hall and can help review and merge

**Comment by mogoodrich** [comment]
Thanks!  Sorry I missed the last day of the hackathon... I am back in Boston and will try to take a look at this today.

**Comment by mogoodrich** [comment]
I took a quick glance and this looks good to me, though honestly I'm not a REST expert and haven't done much with Patient Flags in years... I don't want to be a bottleneck.

What was the final outcome of the hackathon?  Do we forsee someone continuing on with this work further?

**Comment by ssmusoke** [comment]
@maimoonak Please can squash the commits in this PR

File: omod/src/main/java/org/openmrs/module/patientflags/web/rest/PatientFlagFlagResource.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 @Resource(name = RestConstants.VERSION_1 + PatientFlagsRestController.PATIENT_FLAGS_REST_NAMESPACE + "/flag", supportedClass = Flag.class, supportedOpenmrsVersions = {
         "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*" })
-public class PatientFlagResource extends MetadataDelegatingCrudResource<Flag>{
+public class PatientFlagFlagResource extends MetadataDelegatingCrudResource<Flag>{
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/patientflags/web/rest/PatientFlagFlagResource.java]: Why is Flag repeated in the class name?

**Comment by dkayiwa** [comment]
@maimoonak are you still working on this?

File: omod/src/main/java/org/openmrs/module/patientflags/web/rest/PatientFlagFlagResource.java (hunk 1 above)
Comment (RamanKumar2004) [reply omod/src/main/java/org/openmrs/module/patientflags/web/rest/PatientFlagFlagResource.java]: i have reviewed it

**Comment by dkayiwa** [comment]
Closing because of merge conflicts

**Pull request closed without merging**